### PR TITLE
Removal of deprecations for Contour

### PR DESCRIPTION
**Pull request by DHClimber**
## PR summary
This change is related to [[MNT]: Remove 3.7-deprecated API](https://github.com/matplotlib/matplotlib/issues/26865#top)
#26865 

This pull request removes deprecated classes and methods from contour.py and associated contour.pyi

"pre-commit.ci autofix"

## PR checklist

- [ ] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [ ] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [ ] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [ ] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/coding_guide.html#new-features-and-api-changes)
- [ ] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

### CODE REVIEW

**Review by ksunden — CHANGES_REQUESTED** [review]
The code that has been removed is not needed in the removal note, please refer to the deprecation note, which can be copy/pasted and updated to reflect the removal rather than deprecation:

https://github.com/matplotlib/matplotlib/blob/602803912e7294d1882378b5601d19f5a5e7aebc/doc/api/prev_api_changes/api_changes_3.7.0/deprecations.rst?plain=1#L113-L127

Additionally, please rename the file with the PR number rather than `00001`.

File: doc/api/next_api_changes/removals/26907-DCH.rst
Comment (ksunden) [inline doc/api/next_api_changes/removals/26907-DCH.rst:5]:
```suggestion
```

File: doc/api/next_api_changes/removals/26907-DCH.rst
Comment (ksunden) [inline doc/api/next_api_changes/removals/26907-DCH.rst:19]:
```suggestion
```

This is missing the body of that change note

**Review by ksunden — CHANGES_REQUESTED** [review]
Do you feel comfortable squashing the commits to avoid adding and removing the changelog file?

Other than that, just a bit of clean up of the changelog file left

**Comment by DHClimber** [comment]
> Do you feel comfortable squashing the commits to avoid adding and removing the changelog file?
> 
> Other than that, just a bit of clean up of the changelog file left

Thanks for the feedback, I'll try to figure it out. This is my first time and I'm not familiar with squashing, but I'll try.

File: doc/api/next_api_changes/removals/26907-DCH.rst
Comment (ksunden) [inline doc/api/next_api_changes/removals/26907-DCH.rst:11]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
The ``labelFontProps``, ``labelFontSizeList``, and ``labelTextsList``
attributes of `.ContourLabeler` have been removed.  Use the ``labelTexts``
attribute and the font properties of the corresponding text objects instead.
```

This is the part that was missing from the deprecation note

**Comment by ksunden** [comment]
We have some docs on squashing via `git rebase -i`:

https://matplotlib.org/stable/devel/development_workflow.html#rewrite-commit-history

**Comment by ksunden** [comment]
Also remove from the corresponding `.pyi` file, please

**Comment by DHClimber** [comment]
> We have some docs on squashing via `git rebase -i`:
> 
> https://matplotlib.org/stable/devel/development_workflow.html#rewrite-commit-history

Thanks, I think I figured it out.

**Comment by DHClimber** [comment]
It shows one test failing Mypy. I'm having trouble understanding what the issue is, any suggestions on what I need to do to correct?

**Comment by rcomer** [comment]
I agree the Mypy message could be clearer, but I _think_ it’s just that you have removed some methods from the `.py` file that you have not yet removed from the `.pyi` file.

File: doc/api/next_api_changes/removals/26907-DCH.rst
Comment (DHClimber) [reply doc/api/next_api_changes/removals/26907-DCH.rst:11]: Thanks for the help, and I apologize for the mistake. I updated and just now pushed again.

File: doc/api/next_api_changes/removals/26907-DCH.rst
Comment (ksunden) [inline doc/api/next_api_changes/removals/26907-DCH.rst:14]:
```suggestion
attribute and the font properties of the corresponding text objects instead.

```

File: lib/matplotlib/contour.py
Comment (ksunden) [inline lib/matplotlib/contour.py:1874]:
```suggestion
""" % _docstring.interpd.params)

```

**Review by ksunden — CHANGES_REQUESTED** [review]
Just some lint things, if you can sqaush again, that might be ideal as well, though we can also just do that on merge

**Review by ksunden — APPROVED** [review]

**Comment by QuLogic** [comment]
stubtest is still complaining; the entries corresponding to the removed ones need to be removed from the `.pyi` file as well.

**Comment by DHClimber** [comment]
I'm completely lost at this point. I don't understand where all the problems are coming from.

**Comment by ksunden** [comment]
You seem to have done the (roughly) same thing you did previously regarding rebasing.

If you only have the last commit, it looks correct (though I would want CI to run to be sure).

One issue is that main has also updated the `inheritance-diagram` that you had to update, which is causing a conflict.

Unfortunately that is one big long line that makes it hard to tell what the differences are:

It was edited in #26874, which removed the `BrokenBarHCollection`, so both that and `CLabelText` should be removed.

The following should resolve it:

```bash
git rebase --onto upstream/main HEAD~1
# Resolve the conflict noted above, leaving that line in once, the git `>>>>`/`====`/`<<<<` lines removed, etc and both entries to be removed left out
$EDITOR doc/api/artist_api.rst
git add doc/api/artist_api.rst
git rebase --continue
# Edit commit message
git push --force-with-lease
```

If you'd rather we do it, I'm willing, but that is what I believe is needed.

**Comment by DHClimber** [comment]
> e is that main has also updated the

Thanks, I'll give it one more try. Thanks for the patience.

**Comment by QuLogic** [comment]
@DHClimber it appears something messed up here, as you have only a single newline added to one file now. I think the correct commit is f448c0fc73a624edf46806db961392518b365358.

**Comment by DHClimber** [comment]
> @DHClimber it appears something messed up here, as you have only a single newline added to one file now. I think the correct commit is [f448c0f](https://github.com/matplotlib/matplotlib/commit/f448c0fc73a624edf46806db961392518b365358). You can re-fetch that with (assuming you have the `upstream` remote as we recommend):
> 
> ```
> $ git fetch upstream f448c0fc73a624edf46806db961392518b365358:original-depreciated
> ```
> 
> Then you can reset your branch to it:
> 
> ```
> $ git checkout depreciated
> $ git reset --hard original-depreciated
> ```
> 
> and push to replace it here:
> 
> ```
> $ git push --force-with-lease origin depreciated
> ```

Thank you, I have been out of town traveling. I'll try to fix now.

**Comment by DHClimber** [comment]
It seems like correct commit now, but I'm confused what is causing automated tests to fail. If anyone can provide a hint I'm happy to try and correct it.

**Comment by melissawm** [comment]
It looks like you have some conflicts - it may be that something didn't go right on your rebase. Could you try checking again? Here's relevant documentation, hope it helps: https://matplotlib.org/devdocs/devel/development_workflow.html#manage-commit-history

If you need extra help let us know.

**Comment by DHClimber** [comment]
> It looks like you have some conflicts - it may be that something didn't go right on your rebase. Could you try checking again? Here's relevant documentation, hope it helps: https://matplotlib.org/devdocs/devel/development_workflow.html#manage-commit-history
> 
> If you need extra help let us know.

Sorry, I've tried everything I can think of but nothing is making sense. I can't understand where the conflicts are coming from, I'm just shooting in the dark now.

**Comment by ksunden** [comment]
I think I should have cleaned that up, I actually had a branch from the first time I advised a rebase sitting on one of my machines that I had taken care of the conflicts on, so I just rebased that again and pushed it up

**Comment by DHClimber** [comment]
> I think I should have cleaned that up, I actually had a branch from the first time I advised a rebase sitting on one of my machines that I had taken care of the conflicts on, so I just rebased that again and pushed it up

Thanks! I was so confused, didn't even know that was a possibility.

**Comment by QuLogic** [comment]
Can you update the commit message? It currently is an incomplete statement: 'removal of deprecated'. It would be better if it matched this PR title, for example.

**Comment by DHClimber** [comment]
> Can you update the commit message? It currently is an incomplete statement: 'removal of deprecated'. It would be better if it matched this PR title, for example.

Sorry about that, should be corrected now.

File: lib/matplotlib/contour.pyi
Comment (QuLogic) [inline lib/matplotlib/contour.pyi:19]: Remove this line instead.
```suggestion
```

**Review by QuLogic — APPROVED** [review]

**Comment by DHClimber** [comment]
Just following up, is there anything else that needs to be done here?

**Comment by QuLogic** [comment]
We require two approvals before merges, so this is waiting for a second reviewer.

**Review by ksunden — APPROVED** [review]

**Pull request merged**
